vkcs/backup/resource_vkcs_backup_plan.go

```diff
@@ -0,0 +1,533 @@
+					stringvalidator.OneOf([]string{"cloud_servers", "dbaas"}...),
+					stringvalidator.AtLeastOneOf(path.Expressions{
+						path.MatchRoot("provider_id"),
+					}...),
+				},
+			},
+
+			"instance_ids": schema.ListAttribute{
+				ElementType: types.StringType,
+				Required:    true,
+				Description: "List of ids of instances to make backup for",
+			},
+
+			"region": schema.StringAttribute{
+				Computed:    true,
+				Optional:    true,
+				Description: "The `region` to fetch availability zones from, defaults to the provider's `region`.",
+			},
+		},
+		Description: "Manages a backup plan resource.",
+	}
+}
+
+func (r *PlanResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.config = req.ProviderData.(clients.Config)
+}
+
+func (r *PlanResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan PlanResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	region := plan.Region.ValueString()
+	if region == "" {
+		region = r.config.GetRegion()
+	}
+
+	backupClient, err := r.config.BackupV1Client(region, r.config.GetTenantID())
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating VKCS backup client", err.Error())
+		return
+	}
+
+	providerInfo, err := findProvider(backupClient, plan.ProviderID.ValueString(), plan.ProviderName.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vkcs_backup_plan", err.Error())
+		return
+	}
+
+	resourcesInfo, err := getResourcesInfo(r.config, region, plan.InstanceIDs, providerInfo.Name)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vkcs_backup_plan", err.Error())
+		return
+	}
+
+	planCreateOpts := plans.CreateOpts{
+		Name:       plan.Name.ValueString(),
+		Resources:  resourcesInfo,
+		ProviderID: providerInfo.ID,
+	}
+
+	incrementalBackups := plan.IncrementalBackup.ValueBool()
+	if incrementalBackups {
+		fullDay, err := expandFullDay(plan)
+		if err != nil {
+			resp.Diagnostics.AddError("Error creating vkcs_backup_plan", err.Error())
+			return
+		}
+		planCreateOpts.FullDay = fullDay
+	}
+
+	if plan.FullRetention != nil {
+		planCreateOpts.RetentionType = RetentionFull
+	}
+	if plan.GFSRetention != nil {
+		planCreateOpts.RetentionType = RetentionGFS
+		planCreateOpts.GFS = expandGFS(plan)
+	}
+
+	planResp, err := plans.Create(backupClient, &plans.PlanCreate{Plan: &planCreateOpts}).Extract()
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vkcs_backup_plan", err.Error())
+	}
+	planID := planResp.ID
+	resp.State.SetAttribute(ctx, path.Root("id"), planID)
+
+	triggerCreateOpts := triggers.CreateOpts{
+		Name:   fmt.Sprintf("%s_trigger", plan.Name.ValueString()),
+		PlanID: planID,
+	}
+	triggerProperties := triggers.PropertiesOpts{}
+
+	triggerSchedule, err := expandTriggerSchedule(plan)
+	if err != nil {
```

Comment:
should not we prepare all options before the first creation?

Reply:
Rewrote to prepare as much opts as we can